homeassistant/components/thethingsnetwork/config_flow.py

```diff
@@ -0,0 +1,109 @@
+"""The Things Network's integration config flow."""
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from ttn_client import TTNAuthError, TTNClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import CONF_API_KEY, CONF_APP_ID, CONF_HOSTNAME, DOMAIN, TTN_API_HOSTNAME
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TTNFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize flow."""
+        self._reauth_entry: ConfigEntry | None = None
```

Comment:
This can be set outside of the constructor

Reply:
moved